### ISSUE TITLE
Feature request: -silent switch

### ISSUE THREAD

**Issue by dertuxmalwieder** [fcorbelli/zpaqfranz#25]
Sometimes, even `-summary` is too much. I'd like to see a `-silent` switch that only echoes anything if an error has occurred.

**Comment by fcorbelli** [comment]
-summary is NOT less verbose, but shows other infos (... the... summary)...  
It is a 
-pakka is less verbose than "standard"  
-debug turn on about everything  
The "normal" (default) is rather brief, -verbose is better
-noeta turn off ETA % (script files)   
Your request is too general. More specific is needed. 
_In the command x with parameters z, y this information is shown which is excessive (**WHERE?**)_  

There is output (which I have reduced, because I don't like it) on stderr.   
Why? Because redirection of stdout and stderr sucks  
Basically I could change the messages on stderr (with -silent) so that redirecting to a file disappears everything except errors

**Comment by dertuxmalwieder** [comment]
> Your request is too general. More specific is needed.

I'd like to have a switch to have *no* output *at all*.

**Comment by fcorbelli** [comment]
zpaqfranz (...) >/dev/null
zpaqfranz (...) >null (windows)
You will get just about the total time and result
Too much?

**Comment by dertuxmalwieder** [comment]
For use with scripting, *any* output can be too much. :) 
(Of course, it's only a *suggestion* - feel free to deny it.)

**Comment by fcorbelli** [comment]
In fact I do not agree.
On scripting EVERY output is good, to be compressed and send by e-mail as report
ETA (aka: multiple line with \r) is not good. And, in fact... *-noeta* switch is one of the first change in zpaq I made  
If you do not want output... just redirect to null!
Output on error is much, much shorter (in zpaqfranz) vs zpaq  

_Translations: zpaq write on stderr even non-errors. Why I have maintained this behavior in certain cases, in particular the last, the time? Because even redirecting everything to null I want (at least) global time and the overall result_  

If there was an actual "stream" mode (stdin-stdout) I would agree with you.
A -silent switch wouldn't be easier to remember than a >null

**Comment by dertuxmalwieder** [comment]
I see ... :-)

... closing. Thank you.

**Comment by dertuxmalwieder** [comment]
Stop confusing me!

**Comment by fcorbelli** [comment]
For debugging purposes it is perfectly normal for me to redirect the output (almost) everytime. 
 
So it is the switch **-out** that interests me to evolve more  
But, to make sure I sent EVERYTHING (on the log file), I need ZERO console output, here is **-silent**  
If, using -silent -out, nothing is on the screen, everything will be on file
Yes, I am very, veeery lazy, I used "search and replace" to implement -out
:-)

**Comment by dertuxmalwieder** [comment]
It would be easier to refactor if it would not come as one big cpp file. :-p

**Comment by fcorbelli** [comment]
This would require a makefile, and a portable makefile is often more difficult than a program  
Sometimes make just doesn't exist  
Refactoring is limited by design choices that are extremely difficult to change.  
The "aesthetic" aspects are trivial, the functional ones... not
Choices I didn't make  
Nothing prevents you from doing your fork :-D

**Comment by fcorbelli** [comment]
-silent is now (55.10) present

**Comment by dertuxmalwieder** [comment]
✊

will update the OpenBSD port on Sunday, probably. Currently out of town.

**Comment by fcorbelli** [comment]
Relax.  
I am dealing with a NON-Intel CPU user (in the issues), the upcoming release have a fix for... Linux, not BSD :)
